utils/common/src/main/java/org/apache/brooklyn/util/http/HttpExecutorFactory.java

```diff
@@ -0,0 +1,65 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.brooklyn.util.http;
+
+import java.util.Map;
+import java.util.Map.Entry;
+
+import org.apache.brooklyn.util.collections.MutableMap;
+import org.apache.brooklyn.util.exceptions.Exceptions;
+import org.apache.brooklyn.util.http.executor.HttpExecutor;
+import org.apache.brooklyn.util.javalang.Reflections;
+import org.apache.brooklyn.util.text.StringPredicates;
+import org.apache.brooklyn.util.text.Strings;
+
+import com.google.common.base.Optional;
+import com.google.common.collect.Iterables;
+
+public class HttpExecutorFactory {
+    public static final String HTTP_EXECUTOR_CLASS = "httpExecutorClass";
+
+    public static final String HTTP_EXECUTOR_CLASS_PROPERTIES_PREFIX = HTTP_EXECUTOR_CLASS + ".";
+
+    private HttpExecutorFactory() {
+        
+    }
+
+    public <T, S> HttpExecutor getHttpExecutor(Map<?, ?> props) {
+        HttpExecutor httpExecutor;
+        
+        Optional<Map<?,?>> executorProps =  (Optional<Map<?, ?>>) Iterables.tryFind(props.keySet(), StringPredicates.isStringStartingWith(HTTP_EXECUTOR_CLASS_PROPERTIES_PREFIX));
+
+        if (executorProps.isPresent()) {
+            Map<?, ?> instanceProps = executorProps.get();
+
+            String httpExecutorClass = (String) instanceProps.get(HTTP_EXECUTOR_CLASS);
+            Map<String, String> httpExecutorProps = MutableMap.of();
+            for (Entry<?, ?> entry: instanceProps.entrySet()) {
+                String keyName = Strings.removeFromStart((String)entry.getKey(), HTTP_EXECUTOR_CLASS_PROPERTIES_PREFIX);
+                httpExecutorProps.put(keyName, (String)entry.getValue());
+            }
+
+            Reflections reflections = new Reflections(HttpExecutorFactory.class.getClassLoader()); 
```

Comment:
Better use `ClassLoaderUtils` instead of `Reflections` (here and elsewhere in the PR) to be able to use some classloading goodies in OSGi land.